dask/array/core.py

```diff
@@ -2842,12 +2842,10 @@ def where(condition, x=None, y=None):
     x = broadcast_to(x, shape).astype(dtype)
     y = broadcast_to(y, shape).astype(dtype)
 
-    if isinstance(condition, (bool, np.bool8)):
-        if condition:
-            return x
-        else:
-            return y
+    if np.isscalar(condition):
+        return x if condition else y
     else:
+        condition = asarray(condition).astype('bool')
         return choose(condition, [y, x])
```

Comment:
Was wondering the same thing a while back when I was looking at the code, tried changing the code using the patch shown below and did not see substantial differences in time. Using the test code shown below, the run time was `~360us`.

As a side note, the test cases that you have provided show only 1 run each. Though admittedly they use a lot of data, it would be nice to know if that difference between `where` and `choose` is maintained after repeated runs.

<br>

Test Code:

<details>

```python
import dask.array as da

a = da.zeros(100000000, chunks=10000000)
b = da.ones(100000000, chunks=10000000)
c = da.random.randint(0, 2, 100000000, chunks=10000000)
```

</details>

<br>

Patch to use `where` directly:

<details>

```diff
diff --git a/dask/array/core.py b/dask/array/core.py
index 2371200..978b563 100644
--- a/dask/array/core.py
+++ b/dask/array/core.py
@@ -2848,7 +2848,7 @@ def where(condition, x=None, y=None):
         else:
             return y
     else:
-        return choose(condition, [y, x])
+        return elemwise(np.where, condition, x, y)
 
 
 @wraps(chunk.coarsen)
```

</details>